### PR TITLE
Merge Roslyn branch to Master

### PR DESCRIPTION
**Pull request by OmarTawfik**
@Microsoft/fsharp-compiler @jaredpar @cyrusnajmabadi

### CODE REVIEW

File: src/fsharp/FSharp.Compiler/InternalsVisibleTo.fs [hunk 1]
```diff
@@ -17,6 +17,7 @@ open System.Reflection
...
 [<assembly:System.Runtime.CompilerServices.InternalsVisibleTo("FSharp.Compiler.Unittests, PublicKey=002400000480000094000000060200000024000052534131000400000100010007D1FA57C4AED9F0A32E84AA0FAEFD0DE9E8FD6AEC8F87FB03766C834C99921EB23BE79AD9D5DCC1DD9AD236132102900B723CF980957FC4E177108FC607774F29E8320E92EA05ECE4E821C0A5EFE8F1645C4C0C93C1AB99285D622CAA652C1DFAD63D745D6F2DE5F17E5EAF0FC4963D261C8A12436518206DC093344D5AD293")>]
+[<assembly:System.Runtime.CompilerServices.InternalsVisibleTo("FSharp.Editor, PublicKey=002400000480000094000000060200000024000052534131000400000100010007D1FA57C4AED9F0A32E84AA0FAEFD0DE9E8FD6AEC8F87FB03766C834C99921EB23BE79AD9D5DCC1DD9AD236132102900B723CF980957FC4E177108FC607774F29E8320E92EA05ECE4E821C0A5EFE8F1645C4C0C93C1AB99285D622CAA652C1DFAD63D745D6F2DE5F17E5EAF0FC4963D261C8A12436518206DC093344D5AD293")>]
```
Comment (Pilchie) [inline src/fsharp/FSharp.Compiler/InternalsVisibleTo.fs]: Why do we need this?

Comment (OmarTawfik) [inline src/fsharp/FSharp.Compiler/InternalsVisibleTo.fs]: @Pilchie we need to access Range and Span types in the compiler to convert back to Roslyn TextSpan and TextLinePosition.

**Comment by KevinRansom** [comment]
Looks good to me.

File: src/fsharp/FSharp.Compiler/InternalsVisibleTo.fs (hunk 1 above)
Comment (Pilchie) [inline src/fsharp/FSharp.Compiler/InternalsVisibleTo.fs]: Granting IVT the compiler is a pretty big hammer for that - seems like making those types public might be preferable, but I don't really know the policy for the F# codebase.

**Pull request merged**